app/scripts/modules/core/src/presentation/forms/inputs/expression/ExpressionInput.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import { truncate, isEqual } from 'lodash';
+import { parseSpelExpressions } from './spel2js.templateParser';
+
+export interface ISpelError {
+  message: string;
+  context: string;
+  contextTruncated: string;
+}
+
+export interface IExpressionChange {
+  value: string;
+  spelError: ISpelError;
+  spelPreview: string;
+}
+
+export interface IExpressionInputProps {
+  value: string;
+  onChange: (evt: React.ChangeEvent<HTMLInputElement>) => void;
+  onExpressionChange: (expressionChange: IExpressionChange) => void;
+  placeholder?: string;
+  context?: object;
+}
+
+export class ExpressionInput extends React.Component<IExpressionInputProps> {
```

Comment:
An example of a more complex "Input".  It is a controlled input (meaning the state is managed by the parent cmponent). It renders a text input, evaluates the expression using the javascript SpEL library, and emits the results to the parent component using a callback prop.

This input is then used by the `ExpressionField` and `ExpressionRegexField`.